src/lib/stores/investments-view.svelte.ts

```diff
@@ -0,0 +1,66 @@
+import type { InvestmentWithColorIndex } from '$lib/types'
+import { SvelteMap } from 'svelte/reactivity'
+
+export interface InvestmentsViewStore {
+	allInvestments: InvestmentWithColorIndex[]
+	toggleHide: (id: number) => void
+	toggleFocus: (id: number) => void
+	isHidden: (id: number) => boolean
+	isFocused: (id: number) => boolean
+}
+
+export function withInvestmentsViewStore(
+	investments: InvestmentWithColorIndex[],
+): InvestmentsViewStore {
+	let allInvestments = $state<InvestmentWithColorIndex[]>(investments)
+	const hiddenInvestments = $state<SvelteMap<number, boolean>>(new SvelteMap())
+	const visibleInvestments = $derived(
+		allInvestments.filter((inv) => !hiddenInvestments.has(inv.id)),
+	)
+	const focusedInvestment = $derived(
+		visibleInvestments.length === 1 ? visibleInvestments[0] : undefined,
+	)
+
+	function toggleHide(id: number) {
+		if (hiddenInvestments.has(id)) {
+			hiddenInvestments.delete(id)
+		} else {
+			hiddenInvestments.set(id, true)
+		}
+	}
+
+	function toggleFocus(id: number) {
+		if (hiddenInvestments.size === allInvestments.length - 1 && !hiddenInvestments.has(id)) {
+			reset()
+		} else {
+			reset()
+			allInvestments.map((inv) => {
```

Comment:
`map` creates a copy of the array which we don't use at all here
```suggestion
			allInvestments.forEach((inv) => {
```